x/concentrated-liquidity/query.go

```diff
@@ -17,54 +17,77 @@ import (
 	types "github.com/osmosis-labs/osmosis/v20/x/concentrated-liquidity/types"
 )
 
+const invalidTickIndex = int64(-1)
+
 // This file contains query-related helper functions for the Concentrated Liquidity module
 
-// GetTickLiquidityForFullRange returns an array of liquidity depth for all ticks existing from min tick ~ max tick.
-func (k Keeper) GetTickLiquidityForFullRange(ctx sdk.Context, poolId uint64) ([]queryproto.LiquidityDepthWithRange, error) {
+// GetTickLiquidityForFullRange returns a slice of liquidity buckets for all tick ranges existing from min tick ~ max tick.
+// Returns index of the bucket that corresponds to the current tick.
+// For cases where there is no liqudity in the bucket but there may be liquidity to the right, the value will be -1.
+// For cases where there is no liquidity in the bucket but there may be liquidity to the left, the value will be len(liquidityDepthsForRange).
+// Otherwise, the index points to the bucket that corresponds to the current tick.
```

Comment:
It might be useful to more concretely define "bucket" vs. "tick" either here or in docs

Reply:
The docs defines terminologies for these AFAIR!

Reply:
https://github.com/osmosis-labs/osmosis/blob/f721f73fa3afeaa16b39c9da89b25cdac4c98ee4/x/concentrated-liquidity/README.md?plain=1#L1893-L1895